### ISSUE TITLE
Unloading plugins conditionally after iniatialized/enabled on startup.

### ISSUE THREAD

**Issue by Mexflubber** [DavidWells/analytics#433]
Hi,

When my page loads I enable my google-analytics, facebook-ads, intercom, posthog, but under certain conditions I want to disable them. I am doing the following.

``` javascript
analytics.plugins
        .disable(
          [
            'google-analytics',
            'google-tag-manager',
            'facebook-ads',
            'intercom',
            'posthog',
          ],
          () => {
            console.log('Disabled plugins')
          },
        )
        .catch((error) => {
          console.error('Error disabling plugins', error)
        })
```

But even after disabling it I am seeing network calls to all services and seeing the event reach the analytics services. At last for the page() event.

I do get the "Disabled plugins" output in my console (callback function) and there's no "Error disabling plugins" being printed.

Thanks,
Aldo

**Comment by DavidWells** [comment]
Once the scripts load on the page they are entirely out of the libraries control if they are doing stuff themselves.

Check `analytics.getState().plugins` to verify the plugins are disabled. Then you will need to see what the actual scripts the plugins load are doing. I think the only way to "patch" this is by monkey patching the methods they create (if possible) when you disable them in your callback (where `console.log('Disabled plugins')` is). Or maybe even deleting the scripts themselves from DOM and killing their globals. 

When plugins are disabled, they will no longer invoke the .page, track, etc calls where they manually call methods from the loaded 3rd party scripts

**Comment by DavidWells** [comment]
TLDR; you can't really initialize third party scripts and reliably remove them from the DOM without a hard reload.

I'd suggest starting the plugins as disabled, then enabling them when your condition is met. E.g. flipping whatever logic you are currently using to disable. This way you can be sure the third party script doesn't do anything you dont want.

**Comment by Mexflubber** [comment]
Thank you ! For some reason I expected them to get disabled even after being loaded initially !
